content/blog/2023/2023-05-18-operators-in-typescript.md

```diff
@@ -0,0 +1,177 @@
+
+~~~ts
+let age = 30;
+let message = "I am " + age + " years old.";
+console.log(message); 
+~~~
+Output:
+~~~ts
+I am 30 years old.
+~~~
+
+In this example, the concatenation operator combines the string "I am " with the value of the age variable (30) and the string " years old." to create the final message.
+
+Concatenation operators are useful in situations where we need to build dynamic strings based on values or user input. By using concatenation operators, we can create custom messages or outputs that are tailored to our specific needs.
+
+## Arithmetic Operators
+Arithmetic operators allow us to perform mathematical operations such as addition, subtraction, multiplication, division etc on numerical values (constants and variables). Let’s take a look at them:
+
+~~~ts
+let x = 5;
+let y = 10;
+
+console.log(x + y); // Output: 15
+console.log(x - y); // Output: -5
+console.log(x * y); // Output: 50
+console.log(x / y); // Output: 0.5
+console.log(y % x); // Output: 0
+
+let z = 3;
+z++; 
+console.log(z);   // Output: 4
+
+let a = 10;
+a--;
+console.log(a);   // Output: 9
+~~~
+
+**Addition (`+`):** adds two or more values
+
+**Subtraction (`-`):** subtracts two or more values
+
+**Multiplication (`*`):** multiplies two or more values
+
+**Division (`/`):** divides two or more values
+
+**Modulus (`%`):** returns the remainder of a division operation
+
+**Increment (`++`):** increases the value of the variable by one
+
+**Decrement (`--`):** decreases the value of the variable by one
+
+## Relational Operators
+Relational Operators are used to compare two values and determine their relationship. Let’s take a look at some relational operators commonly used in Typescript:
+
+~~~ts
+let x = 10;
+let y = 5;
+
+console.log(x == y); // false
+console.log(x === "10"); // false (different data types)
+console.log(x != y); // true
+console.log(x !== "10"); // true (different data types)
+console.log(x > y); // true
+console.log(x < y); // false
+console.log(x >= y); // true
+console.log(x <= y); // false
+~~~
+
+**Equality Operator (`==`):** This operator compares two values but doesn't consider their data types. If the values are equal, it returns true. Otherwise, it returns false.
+
+**Strict Equality Operator (`===`):** This operator compares two values for equality, and it considers their data types. If the values are equal in value and type, it returns true. Otherwise, it returns false.
+
+**Inequality Operator (`!=`):** This operator compares two values for inequality. If the values are not equal, it returns true. Otherwise, it returns false.
+
+**Strict Inequality Operator (`!==`):** This operator compares two values for inequality, and it considers their data types. If the values are not equal in value or type, it returns true. Otherwise, it returns false.
+
+**Greater Than Operator (`>`):** This operator checks if the left operand is greater than the right operand. If it is true, it returns true. Otherwise, it returns false.
```

Comment:
What happens if you use this on strings or other non-numeric data types? Same for the other operators below